src/components/rangeslider/defaults.js

```diff
@@ -12,14 +12,13 @@ var Lib = require('../../lib');
 var attributes = require('./attributes');
 
 
-module.exports = function supplyLayoutDefaults(layoutIn, layoutOut) {
+module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, axName, counterAxes) {
 
-    if(!layoutIn.xaxis || !layoutOut.xaxis) return;
+    if(!layoutIn[axName].rangeslider) return;
```

Comment:
we should be more strict here. `layoutIn[axName].rangeslider` has to be a plain `{}` to be considered visible.